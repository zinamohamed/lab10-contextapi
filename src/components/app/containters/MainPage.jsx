import React, { useEffect, useState } from 'react';
import { getCharacters } from '../../../services/HeyArnoldApi';
import CharacterList from '../presentations/CharacterList';
import { ThemeContext } from '../../../context/ThemeContext';


const MainPage = () => {

  const [characters, setCharacters] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => { 
    getCharacters()
      .then(arnolds => {
        setCharacters(arnolds);
      })
      .finally(() => setLoading(false));
  }, []);

  const { toggle } = React.useContext(ThemeContext); 
  
  return (
    <div style={ toggle ? { background: 'black', color: 'white' } : {}}>
      <CharacterList loading={loading} characters={characters}/>
    </div>
  );

};

export default MainPage;
