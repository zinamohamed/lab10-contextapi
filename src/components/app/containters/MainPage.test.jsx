/** 
* @jest-environment jsdom
*/
import React from 'react';
import { render, screen } from '@testing-library/react';
import  MainPage  from './MainPage';

describe('Testing MainPage Container', () => { 
  it('should render a list of Hey Arnold characters on screen', async () => { 
    render(<MainPage/>);
    
    const ul = await screen.findByRole('list', { name: 'arnold' });
    expect(ul).toMatchSnapshot();
  });
});