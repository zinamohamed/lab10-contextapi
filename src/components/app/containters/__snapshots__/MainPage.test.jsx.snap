// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainPage Container should render a list of Hey Arnold characters on screen 1`] = `
<ul
  aria-label="arnold"
>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
    />
    <h4>
      Arnold Shortman
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
    />
    <h4>
      Grandpa Phil
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/7/7b/DB87AC2A-C2C2-4DAB-AECF-B9F7273C746B.png/revision/latest/scale-to-width-down/306?cb=20190816101257"
    />
    <h4>
      Grandma Gertie
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
    />
    <h4>
      Mitzi
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
    />
    <h4>
      Arnie
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
    />
    <h4>
      Miles Shortman
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fa/Stella.jpg/revision/latest?cb=20101225143956"
    />
    <h4>
      Stella Shortman
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
    />
    <h4>
      Grandpa's father
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
    />
    <h4>
      Grandpa's grandfather
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/c/c5/Oskar_Kokoshka.png/revision/latest/scale-to-width-down/310?cb=20180707032418"
    />
    <h4>
      Oskar Kokoshka
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fe/Suzie.jpg/revision/latest/scale-to-width-down/310?cb=20110223003447"
    />
    <h4>
      Suzie Kokoshka
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
    />
    <h4>
      Nancy
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/6/67/Ernie.jpeg/revision/latest/scale-to-width-down/310?cb=20181213181427"
    />
    <h4>
      Ernie Potts
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/6/60/Mr._Hyunh.png/revision/latest/scale-to-width-down/310?cb=20180707031441"
    />
    <h4>
      Mr. Hyunh
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/8/83/Mai_Hyunh.jpeg/revision/latest/scale-to-width-down/310?cb=20190118192457"
    />
    <h4>
      Mai Hyunh
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/8/80/Mr._Smith.jpg/revision/latest?cb=20110206233402"
    />
    <h4>
      Mr. Smith
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/762C69BE-23BA-48BA-9B14-6AF3246FCD3F.jpeg/revision/latest/scale-to-width-down/304?cb=20190106211607"
    />
    <h4>
      Lana Vail
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/Mr._Purdy_%28character%29.jpeg/revision/latest?cb=20190125214742"
    />
    <h4>
      Mr. Purdy (character)
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
    />
    <h4>
      Helga G. Pataki
    </h4>
  </div>
  <div>
    <img
      alt=""
      src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Big_Bob.jpg/revision/latest/scale-to-width-down/310?cb=20170428021615"
    />
    <h4>
      Robert "Big Bob" Pataki
    </h4>
  </div>
</ul>
`;
